### ISSUE TITLE
LIEF version

### ISSUE THREAD

**Issue by mashalzainab** [sophos-ai/SOREL-20M#30]
Which version of LIEF was used to extract features from the binaries using ember? 
I am trying to replicate the results but by extracting features from test binaries and I have mismatched features.

**Comment by gxenos** [comment]
Use Python 3.6 and the LIEF version in the requirements.txt of EMBER.

**Comment by mashalzainab** [comment]
You mean lief==0.9.0. 
Thanks, I will try. 
Maybe it also differs because I have a smaller test set as compared to the original and the model gives higher scores for benign files.

**Comment by mashalzainab** [comment]
I tried with python 3.6 and lief==0.9.0 and still the model doesn't perform well. This is my code, can you please tell me if I am missing something? 
It gives me high confidence scores for benign files. 



`

    def evaluate_network(test_files_folder, test_labels_file, results_file, checkpoint_file, db_path,evaluate_malware=True,evaluate_count=True, evaluate_tags=True,remove_missing_features='scan'):
        
    print('...loading model')
  
    model = PENetwork(use_malware=True, use_counts=True, use_tags=True, n_tags=len(Dataset.tags),
                      feature_dimension=2381)

    model.load_state_dict(torch.load(checkpoint_file), strict=False)

    model.to(settings.DEVICE)

    extractor = PEFeatureExtractor(2, print_feature_warning=False)
    df = pd.read_csv(test_labels_file)
    shas, features, labels = [], [], {'malware': []}

    for _, row in tqdm.tqdm(df.iterrows(), total=len(df), desc="Extracting Features"):
        sha256, label = row['sha256'], row['is_malware']
        pattern = re.compile(rf"^{sha256}(\.[^.]+)?(\.[^.]+)?$")
        matching_files = [f for f in os.listdir(test_files_folder) if pattern.match(f)]
        file_path = os.path.join(test_files_folder, matching_files[0]) if matching_files else None

        if not file_path or not os.path.exists(file_path):
            raise FileNotFoundError(f"No matching file found for {sha256} in {test_files_folder}")

        try:
            with open(file_path, "rb") as f:
                bytez = f.read() 
            features.append(extractor.feature_vector(bytez))
            shas.append(sha256)
            labels['malware'].append(label)
        except Exception as e:
            print(f"Error processing {sha256}: {e}")

    
    #print(features)

    features_tensor = torch.tensor(np.array(features)).to(settings.DEVICE)
    labels_tensor = {key: torch.tensor(value) for key, value in labels.items()}

    predictions = model(features_tensor)
    results = normalize_results(labels_tensor, predictions, use_count=False, use_tags=False)
    pd.DataFrame(results, index=shas).to_csv(results_file, header=True, index_label="sha256")

    print('...done') `

**Comment by mashalzainab** [comment]
I was able to do it by changing the LMDBReader class of the dataset.py file. And used the same rest of the code. 


`class LMDBReader(object): 

    def __init__(self, path, postproc_func=None, test_files_folder=None):
        self.env = lmdb.open(path, readonly=True, lock=False, map_size=1e13, max_readers=1024)
        self.postproc_func = postproc_func 
        self.test_files_folder = test_files_folder 

    def __call__(self, key, extract_from_files=False): 

        if extract_from_files: 
            pattern = re.compile(rf"^{key}(\.[^.]+)?(\.[^.]+)?$")

            # Search for matching file in the folder
            matching_files = [f for f in os.listdir(self.test_files_folder) if pattern.match(f)] 

            file_path = os.path.join(self.test_files_folder, matching_files[0]) if matching_files else None


            # Ensure file_path is valid before using it
            if file_path is None:
                raise FileNotFoundError(f"No matching file found for {key} in {self.test_files_folder}") 
            # Ensure the file exists
            if os.path.exists(file_path): 
                try:
                    extractor = PEFeatureExtractor(2, print_feature_warning=False)
                    with open(file_path, "rb") as f:
                        bytez = f.read()  
                    feature = extractor.feature_vector(bytez)  
                    x = {0 : feature.flatten().tolist()}
                    if self.postproc_func is not None:
                        x = self.postproc_func(x) 
                    return x 
                except Exception as e:
                    print(f"Error processing {key}: {e}")

        else: 
            with self.env.begin() as txn:
                x = txn.get(key.encode('ascii')) 
            if x is None:return None
            x = msgpack.loads(zlib.decompress(x),strict_map_key=False) 
            #print(x)
            if self.postproc_func is not None:
                x = self.postproc_func(x)
            return x
`
